lib/Conversion/TritonGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -103,11 +103,59 @@ struct LocalDeallocOpConversion
   }
 };
 
+struct LocalStoreOpConversion
+    : public ConvertOpToLLVMPattern<triton::gpu::LocalStoreOp> {
+public:
+  using ConvertOpToLLVMPattern<
+      triton::gpu::LocalStoreOp>::ConvertOpToLLVMPattern;
+
+  LocalStoreOpConversion(const LLVMTypeConverter &converter,
+                         const TargetInfoBase &targetInfo,
+                         PatternBenefit benefit = 1)
+      : ConvertOpToLLVMPattern<triton::gpu::LocalStoreOp>(converter, benefit),
+        targetInfo(targetInfo) {}
+
+  LogicalResult
+  matchAndRewrite(triton::gpu::LocalStoreOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Location loc = op->getLoc();
+    // TODO pawel: getDefiningOp() below can return nullptr. How do we find
+    // shmem base?
+    Value smemBase = LLVM::getSharedMemoryBase(loc, rewriter,
+                                               op.getResult().getDefiningOp());
+    auto srcTy = op.getSrc().getType();
+    auto resultTy = cast<MemDescType>(op.getResult().getType());
+    auto resultShapePerCTA = triton::gpu::getShapePerCTA(resultTy);
+    auto typeConverter = getTypeConverter();
+    auto sharedLayout =
+        cast<triton::gpu::SharedEncodingAttr>(resultTy.getEncoding());
+    auto order = sharedLayout.getOrder();
+
+    auto elemTy = typeConverter->convertType(srcTy.getElementType());
+
+    int32_t elemSize = elemTy.getIntOrFloatBitWidth();
+    unsigned numElems = triton::gpu::getTotalElemsPerThread(srcTy);
+    auto dstStrides = LLVM::getStridesFromShapeAndOrder(resultShapePerCTA,
+                                                        order, loc, rewriter);
+    auto inVals = unpackLLElements(loc, adaptor.getSrc(), rewriter);
+    storeDistributedToShared(op.getSrc(), inVals, dstStrides, op.getResult(),
+                             smemBase, elemTy, loc, rewriter, targetInfo);
+    rewriter.eraseOp(op);
+
```

Comment:
not sure if it is practical but could we re-use directly `lowerDistributedToShared `? Doesn't look like it needs much alloc specific

Reply:
Tell me what you think now.